CMakeLists.txt

```diff
@@ -651,35 +652,67 @@ if(Iconv_FOUND)
 	target_link_libraries(torrent-rasterbar PRIVATE Iconv::Iconv)
 endif()
 
-find_public_dependency(OpenSSL)
-set_package_properties(OpenSSL
+find_public_dependency(GnuTLS)
+set_package_properties(GnuTLS
 	PROPERTIES
-		URL "https://www.openssl.org/"
-		DESCRIPTION "Full-strength general purpose cryptography library"
+		URL "https://www.gnutls.org/"
+		DESCRIPTION "GnuTLS is a free software implementation of the TLS and DTLS protocols"
 		TYPE RECOMMENDED
 		PURPOSE "Provides HTTPS support to libtorrent"
 )
 
-if(OPENSSL_FOUND)
+if(NOT gnutls)
+	find_public_dependency(OpenSSL)
+	set_package_properties(OpenSSL
+		PROPERTIES
+			URL "https://www.openssl.org/"
+			DESCRIPTION "Full-strength general purpose cryptography library"
+			TYPE RECOMMENDED
+			PURPOSE "Provides HTTPS support to libtorrent"
+	)
+	if(OPENSSL_FOUND)
+		# TODO: needed until https://gitlab.kitware.com/cmake/cmake/issues/19263 is fixed
+		if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
+			target_link_libraries(OpenSSL::Crypto INTERFACE crypt32)
+		endif()
 
-	# TODO: needed until https://gitlab.kitware.com/cmake/cmake/issues/19263 is fixed
-	if(WIN32 AND OPENSSL_USE_STATIC_LIBS)
-		target_link_libraries(OpenSSL::Crypto INTERFACE crypt32)
+		target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
+		target_compile_definitions(torrent-rasterbar
+			PUBLIC
+				TORRENT_USE_OPENSSL
+				TORRENT_SSL_PEERS
+				OPENSSL_NO_SSL2)
 	endif()
+endif()
 
-	target_link_libraries(torrent-rasterbar PUBLIC OpenSSL::SSL)
-	target_compile_definitions(torrent-rasterbar
-		PUBLIC
-			TORRENT_USE_LIBCRYPTO
-			TORRENT_USE_OPENSSL
-			TORRENT_SSL_PEERS
-			OPENSSL_NO_SSL2)
-	target_sources(torrent-rasterbar PRIVATE src/pe_crypto)
+if(gnutls OR NOT OPENSSL_FOUND)
```

Comment:
I'm not quite following this. `gnutls` is set by the first call to `find_public_dependency(GnuTLS)`, is it?

If it's set and `OPENSSL_FOUND` is *not* set, then you call `find_public_dependency(GnuTLS)` again? Why does it need to be called twice?

Reply:
are the first calls to `find_public_dependency(GnuTLS)` and `set_package_properties(GnuTLS, ...` lef-overs that aren't needed anymore?

Reply:
No, `gnutls` is actually the cmake option set by the user (disabled by default). So the logic is to use GnuTLS if either set by the user or if OpenSSL is not found.

Reply:
However the second call is a rebase mistake I think, it's indeed useless :facepalm: